winch/codegen/src/isa/x64/asm.rs

```diff
@@ -2082,6 +2226,155 @@ impl Assembler {
         });
     }
 
+    /// Takes the lower lanes of vectors of floats in `src1` and `src2` and
+    /// interleaves them in `dst`.
+    pub fn xmm_vunpcklp_rrm(
+        &mut self,
+        src1: Reg,
+        src2: &Address,
+        dst: WritableReg,
+        size: OperandSize,
+    ) {
+        let op = match size {
+            OperandSize::S32 => AvxOpcode::Vunpcklps,
+            _ => unimplemented!(),
+        };
+
+        let address = Self::to_synthetic_amode(
+            src2,
+            &mut self.pool,
+            &mut self.constants,
+            &mut self.buffer,
+            MemFlags::trusted(),
+        );
+
+        self.emit(Inst::XmmRmiRVex {
+            op,
+            src1: src1.into(),
+            src2: XmmMemImm::unwrap_new(RegMemImm::mem(address)),
+            dst: dst.to_reg().into(),
+        });
+    }
+
+    /// Performs a subtraction on two vectors of floats and puts the results in
+    /// `dst`.
+    pub fn xmm_vsub_rrm(&mut self, src1: Reg, src2: &Address, dst: WritableReg, size: OperandSize) {
+        let op = match size {
+            OperandSize::S64 => AvxOpcode::Vsubpd,
+            _ => unimplemented!(),
+        };
+
+        let address = Self::to_synthetic_amode(
+            src2,
+            &mut self.pool,
+            &mut self.constants,
+            &mut self.buffer,
+            MemFlags::trusted(),
```

Comment:
Is it generally fine to consider those memory access trusted? It seems to me that, in this specific case, this is fine, but the callee cannot assume that address will indeed be aligned and non-trapping, no?

Reply:
> Is it generally fine to consider those memory access trusted?

It is -- as a rule of thumb, any accesses emitted by the compiler as part of loading and storing/loading spilled values are considered trusted.

On the other hand, any accesses emitted by the compiler by virtue of translating any Wasm instruction which explicitly interacts with linear memory is considered untrusted and _must_ comply with the semantics dictated by such instruction, like alignment and trapping.


Practically speaking what that means is that if the operand used in this instruction is an incorrect address, this memory access will likely result in a segmentation fault rather than in a Wasm trap.